src/CFamily/CMake/ActiveConfigProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Newtonsoft.Json.Linq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.Helpers;
+
+namespace SonarLint.VisualStudio.CFamily.CMake
+{
+    internal interface IActiveConfigProvider
+    {
+        /// <summary>
+        /// Returns the name of the current configuration
+        /// </summary>
+        /// <param name="rootDirectory">The root directory for the solution</param>
+        string GetActiveConfig(string rootDirectory);
+    }
+
+    internal class ActiveConfigProvider : IActiveConfigProvider
+    {
+        private const string VSDefaultConfiguration = "x64-Debug";
+        private const string RelativeSettingsPath = ".vs\\ProjectSettings.json";
```

Comment:
Just wondering, is that configurable in any way?

Reply:
I doubt it; the contents of the '.vs' folder are private, so there would no reason to make the file name/location configurable.